Move Catalogue To GafferScene, Reassign TypeId ranges

danieldresser-ie opened the pull request:
As requested this morning, I've grouped these 2 very mechanical prerequisites for picking into one PR.

One commit switches our id ranges to use the new id range from https://github.com/ImageEngine/cortex/pull/1469

Another commit moves Catalogue, CatalogueSelect, and Display from GafferImage to GafferScene. ( I just delete the old type ids from GafferImage - usually we would leave these unused ids in the list, but since we're reworking ids at the same time, I might as well clean that up as well ). Lots of code changes to make this happen, but it's all pretty mechanical, and the tests seem to be passing ( there were a few rounds of catching leftover odds and ends, but hopefully they're all caught now ).

danieldresser-ie commented:
Remembered to glance at the test I'd left running before bed, and noticed a stray import that snuck in during a rebase and broke everything ... fixed that, now everything should actually be working.

johnhaddon reviewed (commented):
Thanks Daniel! It's going to be nice not to have to worry about TypeId exhaustion. Comments inline as usual...

danieldresser-ie commented on include/GafferCycles/TypeIds.h:
```diff
@@ -41,16 +41,16 @@ namespace GafferCycles
+	CyclesAttributesTypeId = 122500,
+	CyclesOptionsTypeId = 122501,
+	CyclesRenderTypeId = 122502, // Available for reuse
+	InteractiveCyclesRenderTypeId = 122503, // Available for reuse
```
Addressed in: [936a417](https://github.com/GafferHQ/gaffer/pull/6434/commits/936a4171762d371b303b8d18e376bf9f6dc0b006)

danieldresser-ie commented on include/GafferScene/TypeIds.h:
```diff
@@ -42,159 +42,159 @@ namespace GafferScene
-	PreviewProceduralTypeId = 110649,
+	PreviewPlaceholderTypeId = 120146,
+	PreviewGeometryTypeId = 120147,
+	PreviewProceduralTypeId = 120148,
```
Added a new commit where I move these to a new header in IECoreScenePreview: [7e6f866](https://github.com/GafferHQ/gaffer/pull/6434/commits/7e6f866c5aefe56d7612fd9883e1b25b5c7ec896)

danieldresser-ie commented on python/GafferScene/CatalogueSelect.py:
```diff
@@ -37,6 +37,7 @@
 
 import Gaffer
 import GafferImage
+import GafferScene
```
Adressed in: [99056b3](https://github.com/GafferHQ/gaffer/pull/6434/commits/99056b32dfb655a5800553d402c3b69d6c3029ef)

danieldresser-ie commented on python/GafferScene/__init__.py:
```diff
@@ -42,8 +42,22 @@
+		return GafferScene.Catalogue
+	Catalogue = property( __getCatalogue )
+sys.modules[__name__].__class__ = ThisMod
+"""
```
Adressed in: [99056b3](https://github.com/GafferHQ/gaffer/pull/6434/commits/99056b32dfb655a5800553d402c3b69d6c3029ef)

danieldresser-ie commented on src/GafferScene/Display.cpp:
```diff
@@ -381,7 +381,18 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 
-} // namespace GafferImage
+// The description above registers this display driver as GafferScene::GafferDisplayDriver. But old scripts
+// will contain references to GafferScene::GafferDisplayDriver - we need to register it as the old name as well.
```
Adressed in: [99056b3](https://github.com/GafferHQ/gaffer/pull/6434/commits/99056b32dfb655a5800553d402c3b69d6c3029ef)

danieldresser-ie commented on src/GafferScene/Display.cpp:
```diff
@@ -381,7 +381,18 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
+{
+	return new GafferDisplayDriver( displayWindow, dataWindow, channelNames, parameters );
+}
+const int gafferDisplayDriverCompatibility = [](){
```
Adressed in: [99056b3](https://github.com/GafferHQ/gaffer/pull/6434/commits/99056b32dfb655a5800553d402c3b69d6c3029ef)

danieldresser-ie commented on startup/GafferImage/catalogueCompatibility.py:
```diff
@@ -0,0 +1,54 @@
+	CatalogueSelect = property( lambda self : PatchedGafferImage._gafferSceneRedirect( "CatalogueSelect" ) )
+	Display = property( lambda self : PatchedGafferImage._gafferSceneRedirect( "Display" ) )
+
+sys.modules["GafferImage"].__class__ = PatchedGafferImage
```
Addressed in: [99056b3](https://github.com/GafferHQ/gaffer/pull/6434/commits/99056b32dfb655a5800553d402c3b69d6c3029ef)

I wasn't sure whether or not to redirect GafferImageUI.CatalogueSelectUI? I ended up not doing it, because I don't see anything in there that anyone could be using.

danieldresser-ie commented on python/GafferSceneTest/InteractiveRenderTest.py:
```diff
@@ -282,7 +282,7 @@ def testAddAndRemoveLocation( self ) :
 
 		s = Gaffer.ScriptNode()
-		s["catalogue"] = GafferImage.Catalogue()
+		s["catalogue"] = GafferScene.Catalogue()
```
Addressed in: [99056b3](https://github.com/GafferHQ/gaffer/pull/6434/commits/99056b32dfb655a5800553d402c3b69d6c3029ef)

danieldresser-ie commented:
Addressed all feedback, added Changelog entry. Not sure whether it was easier to review this before squashing, so still needs a squash, but otherwise, should be good to merge.

johnhaddon commented:
Looks like we have a missing import in the tests :

```
Error: ERROR: testBoxedCatalogue (GafferSceneUITest.CatalogueUITest.CatalogueUITest.testBoxedCatalogue)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/gaffer/gaffer/build/python/GafferSceneUITest/CatalogueUITest.py", line 76, in testBoxedCatalogue
    s["b"]["c"] = GafferScene.Catalogue()
                  ^^^^^^^^^^^
NameError: name 'GafferScene' is not defined
```

Otherwise, LGTM. Could you squash at the same time as fixing the import please?

johnhaddon commented:
Yep, feel free to merge.
